python/ray/data/tests/test_consumption.py

```diff
@@ -241,6 +241,22 @@ def test_schema_lazy(ray_start_regular_shared):
     assert ds._plan.execute()._num_computed() == 0
 
 
+def test_schema_cached(ray_start_regular_shared):
+    def check_schema_cached(ds):
+        schema = ds.schema()
```

Comment:
Should we also check that repeated calls to `schema()` are cached?

Reply:
Ah it is checked under this.